control/trajectory_follower/src/pid_longitudinal_controller.cpp

```diff
@@ -821,31 +833,16 @@ PidLongitudinalController::Motion PidLongitudinalController::keepBrakeBeforeStop
 
 autoware_auto_planning_msgs::msg::TrajectoryPoint
 PidLongitudinalController::calcInterpolatedTargetValue(
-  const autoware_auto_planning_msgs::msg::Trajectory & traj, const geometry_msgs::msg::Pose & pose,
-  const size_t nearest_idx) const
+  const autoware_auto_planning_msgs::msg::Trajectory & traj,
+  const geometry_msgs::msg::Pose & pose) const
 {
   if (traj.points.size() == 1) {
     return traj.points.at(0);
   }
 
-  // If the current position is not within the reference trajectory, enable the edge value.
-  // Else, apply linear interpolation
-  if (nearest_idx == 0) {
-    if (motion_common::calcSignedArcLength(traj.points, pose.position, 0) > 0) {
-      return traj.points.at(0);
-    }
-  }
-  if (nearest_idx == traj.points.size() - 1) {
-    if (
-      motion_common::calcSignedArcLength(traj.points, pose.position, traj.points.size() - 1) < 0) {
-      return traj.points.at(traj.points.size() - 1);
-    }
-  }
-
```

Comment:
This code can be removed since ratio is clamped from 0.0 to 1.0 here
https://github.com/autowarefoundation/autoware.universe/blob/1a635b74999d3313df8fa6d050ca5e2ac5318f4d/control/trajectory_follower/include/trajectory_follower/longitudinal_controller_utils.hpp#L122